controllers/release/adapter.go

```diff
@@ -337,28 +399,61 @@ func (a *adapter) EnsureReleaseProcessingIsTracked() (controller.OperationResult
 	return controller.ContinueProcessing()
 }
 
-// EnsureReleaseProcessingResourcesAreCleanedUp is an operation that will ensure that the resources created for the Release
-// Processing step are cleaned up once processing is finished.
-func (a *adapter) EnsureReleaseProcessingResourcesAreCleanedUp() (controller.OperationResult, error) {
-	if !a.release.HasProcessingFinished() {
+// EnsureReleaseFinalProcessingIsTracked is an operation that will ensure that the Release Final PipelineRun status
+// is tracked in the Release being processed.
+func (a *adapter) EnsureReleaseFinalProcessingIsTracked() (controller.OperationResult, error) {
+	if !a.release.IsFinalPipelineProcessing() || a.release.HasFinalPipelineProcessingFinished() {
 		return controller.ContinueProcessing()
 	}
 
-	pipelineRun, err := a.loader.GetReleasePipelineRun(a.ctx, a.client, a.release)
-	if err != nil && !errors.IsNotFound(err) {
+	pipelineRun, err := a.loader.GetReleasePipelineRun(a.ctx, a.client, a.release, metadata.FinalPipelineType)
+	if err != nil {
 		return controller.RequeueWithError(err)
 	}
+	if pipelineRun != nil {
+		err = a.registerFinalProcessingStatus(pipelineRun)
+		if err != nil {
+			return controller.RequeueWithError(err)
+		}
+	}
 
-	roleBinding, err := a.loader.GetRoleBindingFromReleaseStatus(a.ctx, a.client, a.release)
-	if err != nil && !errors.IsNotFound(err) && !strings.Contains(err.Error(), "valid reference to a RoleBinding") {
+	return controller.ContinueProcessing()
+}
+
+// EnsureReleaseManagedProcessingIsTracked is an operation that will ensure that the Release Managed PipelineRun status
+// is tracked in the Release being processed.
+func (a *adapter) EnsureReleaseManagedProcessingIsTracked() (controller.OperationResult, error) {
+	if !a.release.IsManagedPipelineProcessing() || a.release.HasManagedPipelineProcessingFinished() {
+		return controller.ContinueProcessing()
+	}
+
+	pipelineRun, err := a.loader.GetReleasePipelineRun(a.ctx, a.client, a.release, metadata.ManagedPipelineType)
+	if err != nil {
 		return controller.RequeueWithError(err)
 	}
+	if pipelineRun != nil {
+		err = a.registerManagedProcessingStatus(pipelineRun)
+		if err != nil {
+			return controller.RequeueWithError(err)
+		}
+	}
+
+	return controller.ContinueProcessing()
+}
+
+// EnsureReleaseProcessingResourcesAreCleanedUp is an operation that will ensure that the resources created for the Release
+// Processing step are cleaned up once processing is finished. This exists in conjunction with EnsureFinalizersAreCalled because
+// the finalizers should be removed from the pipelineRuns even if the Release is not marked for deletion for quota reasons.
+func (a *adapter) EnsureReleaseProcessingResourcesAreCleanedUp() (controller.OperationResult, error) {
```

Comment:
This one should probably also check for final pipeline not being finished yet

Reply:
Add in the final pipeline